### ISSUE TITLE
list_result fails with "+" in object name

### ISSUE THREAD

**Issue by torsstei** [IBM-Cloud/sql-query-clients#62]
list_results() fails when responseBodyXMLroot in line 317 is this content:
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?><ListBucketResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Name>twc-covid-19</Name><Prefix>tmp/test-cases-mart/collected=2020-04-23T06:59:05 00:00/country_id=US/data.parquet</Prefix><Marker></Marker><MaxKeys>1000</MaxKeys><Delimiter></Delimiter><IsTruncated>false</IsTruncated></ListBucketResult>
```

Doesn't find any result

**Comment by okoethibm** [comment]
happens because the prefix is used as a query parameter without extra encoding in line 318. if the COS URI contains a plus, that is transformed into a space (+ is an ecoding for space, but only in query params, not in URL paths), so COS is looking for the wrong prefix and finds no objects.

there is probably a similar problem in get_result line 204
can we just use botocore with _get_cos_client directly, like in lin 229ff and get rid of direct requests and XML parsing?

**Comment by torsstei** [comment]
More specifically we should list object as in line 542.

**Comment by torsstei** [comment]
Fixed with https://github.com/IBM-Cloud/sql-query-clients/commit/d90bdab5ce3f816eabd219b2e32f2fbf4cfd8a8a and made available as v0.3.14
